main.go

```diff
@@ -195,26 +214,44 @@ func main() {
 		os.Exit(1)
 	}
 
-	buffer, err := downloadFileToBuffer(url, *concurrency)
+	// create tempfile for downloading to
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("Error getting cwd: %v\n", err)
+		os.Exit(1)
+	}
+	destTemp, err := os.CreateTemp(cwd, dest+".partial")
```

Comment:
Does it make sense to use the `*` notation to allow createTemp to create a unique file for the temp, this means that if somehow something is left on disk afterwards/failure of pget before removal, a new temp file would be created instead of trying to scribble over the top

```suggestion
	destTemp, err := os.CreateTemp(cwd, fmt.Sprintf("%s-*.partial", dest)
```

or even `os.CreateTemp(cwd, "*.partial")` which it more-inline with how browsers do it now, so that filenames are really obscured and not accidentally used until complete. Since we either extract from the reference object for the tempfile or rename it, the name doesn't need to include "dest" if we don't want it.